state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -811,32 +811,35 @@ pub mod pallet {
 				match current_epoch.saturating_sub(call.original_epoch) {
 					// The call is stale, clean up storage.
 					n if n >= 2 => {
-						T::Broadcaster::clean_up_broadcast_storage(call.broadcast_id);
+						T::Broadcaster::expire_broadcast(call.broadcast_id);
 						Self::deposit_event(Event::<T, I>::FailedForeignChainCallExpired {
 							broadcast_id: call.broadcast_id,
 						});
 					},
-					// Previous epoch, signature is invalid. Re-sign and store.
-					1 => {
-						if let Some(threshold_signature_id) =
-							T::Broadcaster::threshold_resign(call.broadcast_id)
-						{
+					// Previous epoch, signature is invalid. Re-sign but don't broadcast.
+					1 => match T::Broadcaster::re_sign_broadcast(call.broadcast_id, false) {
+						Ok(threshold_signature_id) => {
 							Self::deposit_event(Event::<T, I>::FailedForeignChainCallResigned {
 								broadcast_id: call.broadcast_id,
 								threshold_signature_id,
 							});
 							FailedForeignChainCalls::<T, I>::append(current_epoch, call);
-						} else {
-							// We are here if the Call needs to be resigned, yet no API call data is
-							// available to use. In this situation, there's nothing else that can be
-							// done.
-							log::error!("Foreign Chain Call message cannot be re-signed: Call data unavailable. Broadcast Id: {:?}", call.broadcast_id);
-						}
+						},
+						Err(e) => {
+							// This can happen if a broadcast is still pending despite
+							// since the previous epoch.
+							// TODO: make sure this can't happen.
```

Comment:
this case could happen theoretically since the braodcast timeout seems to be 10 mins and for a broadcast to be aborted, it will take more than 150*10 mins which turns out to be 25 hours (a little more than 1 epoch time).

Reply:
Yes I thought so but I wasn't sure if for example the pending ccm would prevent the rotation from completing (due to the broadcast barrier).

Reply:
you are right, its true that broadcast barrier will stop this from happening but relying on broadcast barrier is a bit problematic since its chain specific implementation can be different. For example, there can be a chain where there is no broadcast barrier.

Reply:
I agree we should fix this but will leave it out of scope for this PR.
For now the only chains supporting CCM are EVM and have the same restrictions, so it's ok. I think the correct solution is to trigger re-signing of these calls when we rotate, but this is a larger piece of work... I'll open a Linear ticket.